<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.70">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="scvi-tools Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="scvi-tools Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-141905405-3","auto"),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css"><title data-react-helmet="true">scvi-tools 0.9.0 release | scvi-tools</title><meta data-react-helmet="true" property="og:title" content="scvi-tools 0.9.0 release | scvi-tools"><meta data-react-helmet="true" name="description" content="Today we officially released scvi-tools version 0.9.0 (changelog). This release marks the culmination of five months of work on the backend of the codebase, which came after three months of work on the frontend."><meta data-react-helmet="true" property="og:description" content="Today we officially released scvi-tools version 0.9.0 (changelog). This release marks the culmination of five months of work on the backend of the codebase, which came after three months of work on the frontend."><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_tag" content="default"><link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico"><link rel="stylesheet" href="/styles.4e7412cc.css">
<link rel="stylesheet" href="/main.b90caea9.css">
<link rel="preload" href="/styles.3d86d984.js" as="script">
<link rel="preload" href="/runtime~main.de082324.js" as="script">
<link rel="preload" href="/main.0990fd90.js" as="script">
<link rel="preload" href="/1.28b4df87.js" as="script">
<link rel="preload" href="/2.246fa4d6.js" as="script">
<link rel="preload" href="/3.5be6f89c.js" as="script">
<link rel="preload" href="/ccc49370.122b310d.js" as="script">
<link rel="preload" href="/e5436e3b.fe8d3470.js" as="script">
<link rel="preload" href="/c534381f.36cfbbd2.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_11B0">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">scvi-tools</strong></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link" href="/get_started">Get Started</a><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Docs</a><ul class="dropdown__menu"><li><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" class="dropdown__link">Full documentation</a></li><li><a href="https://docs.scvi-tools.org/en/stable/user_guide/user.html" target="_blank" rel="noopener noreferrer" class="dropdown__link">User guide</a></li><li><a href="https://docs.scvi-tools.org/en/stable/user_guide/developer.html" target="_blank" rel="noopener noreferrer" class="dropdown__link">Developer guide</a></li><li><a href="https://docs.scvi-tools.org/en/stable/api/index.html" target="_blank" rel="noopener noreferrer" class="dropdown__link">API reference</a></li></ul></div><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a class="navbar__item navbar__link">About</a><ul class="dropdown__menu"><li><a class="dropdown__link" href="/team">Team</a></li><li><a class="dropdown__link" href="/press">Press</a></li><li><a class="dropdown__link" href="/ecosystem">Ecosystem</a></li><li><a class="dropdown__link" href="/news">News</a></li></ul></div><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/blog">Blog</a><a href="https://discourse.scvi-tools.org/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Discussion</a><a href="https://github.com/YosefLab/scvi-tools" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub</a></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">scvi-tools</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/get_started">Get Started</a></li><li class="menu__list-item menu__list-item--collapsed"><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" role="button" class="menu__link menu__link--sublist">Docs</a><ul class="menu__list"><li class="menu__list-item"><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" class="menu__link">Full documentation</a></li><li class="menu__list-item"><a href="https://docs.scvi-tools.org/en/stable/user_guide/user.html" target="_blank" rel="noopener noreferrer" class="menu__link">User guide</a></li><li class="menu__list-item"><a href="https://docs.scvi-tools.org/en/stable/user_guide/developer.html" target="_blank" rel="noopener noreferrer" class="menu__link">Developer guide</a></li><li class="menu__list-item"><a href="https://docs.scvi-tools.org/en/stable/api/index.html" target="_blank" rel="noopener noreferrer" class="menu__link">API reference</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a role="button" class="menu__link menu__link--sublist">About</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/team">Team</a></li><li class="menu__list-item"><a class="menu__link" href="/press">Press</a></li><li class="menu__list-item"><a class="menu__link" href="/ecosystem">Ecosystem</a></li><li class="menu__list-item"><a class="menu__link" href="/news">News</a></li></ul></li><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/blog">Blog</a></li><li class="menu__list-item"><a href="https://discourse.scvi-tools.org/" target="_blank" rel="noopener noreferrer" class="menu__link">Discussion</a></li><li class="menu__list-item"><a href="https://github.com/YosefLab/scvi-tools" target="_blank" rel="noopener noreferrer" class="menu__link">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper blog-wrapper"><div class="container margin-vert--lg"><div class="row"><div class="col col--2"><div class="sidebar_SWld thin-scrollbar"><h3 class="sidebarItemTitle_Km2m">Recent posts</h3><ul class="sidebarItemList_3UpA"><li class="sidebarItem_2T0D"><a aria-current="page" class="sidebarItemLink_v5H9 sidebarItemLinkActive_1anX" href="/blog/v090">scvi-tools 0.9.0 release</a></li><li class="sidebarItem_2T0D"><a class="sidebarItemLink_v5H9" href="/blog/autotune">Hyperparameter search for scVI</a></li><li class="sidebarItem_2T0D"><a class="sidebarItemLink_v5H9" href="/blog/zero-inflation">Should we zero-inflate scVI?</a></li></ul></div></div><main class="col col--8"><article><header><h1 class="margin-bottom--sm blogPostTitle_3-lP">scvi-tools 0.9.0 release</h1><div class="margin-vert--md"><time datetime="2021-03-03T00:00:00.000Z" class="blogPostDate_Ta7i">March 3, 2021  · 7 min read</time></div><div class="avatar margin-vert--md"><div class="avatar__intro"><h4 class="avatar__name"><a target="_blank" rel="noreferrer noopener">Adam Gayoso, Romain Lopez, Galen Xing, Nir Yosef</a></h4><small class="avatar__subtitle"></small></div></div></header><section class="markdown"><p>Today we officially released <code>scvi-tools</code> version 0.9.0 (<a href="https://docs.scvi-tools.org/en/stable/release_notes/index.html" target="_blank" rel="noopener noreferrer">changelog</a>). This release marks the culmination of five months of work on the backend of the codebase, which came after three months of work on the frontend.
In this short note, we officially introduce <code>scvi-tools</code> as a readily usable codebase that contains many implementations of probabilistic single-cell omics methods, and also features a high-level interface to accelerate the model development process. We start with some historical notes about our previous codebase, which was mostly used for internal developments in the last three years. We then describe the obstacles we found to its external adoption, and the foundational idea behind the new <code>scvi-tools</code> work: a high-level deep probabilistic programming library specialized for single-cell omics data.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="taking-a-step-back-the-original-scvi-codebase"></a>Taking a step back: the original <code>scvi</code> codebase<a class="hash-link" href="#taking-a-step-back-the-original-scvi-codebase" title="Direct link to heading">#</a></h2><p>Many members of the Yosef Lab, and in particular Jeff Regier, Edouard Melhman, Romain, and Adam helped conceive, develop, and maintain the <code>scvi</code> codebase. The initial philosophy was to make the code available for users to run <a href="https://www.nature.com/articles/s41592-018-0229-2" target="_blank" rel="noopener noreferrer">scVI</a>, but also to have a proper codebase for developing novel algorithms for single-cell omics analysis. Over the last three years, we have hosted seven visiting graduate students who wrote their Master&#x27;s theses in the Yosef Lab by building new functionalities, as well as new algorithms (including <a href="https://www.embopress.org/doi/full/10.15252/msb.20209620" target="_blank" rel="noopener noreferrer">scANVI</a>, <a href="https://www.biorxiv.org/content/biorxiv/early/2019/10/10/794875.full.pdf" target="_blank" rel="noopener noreferrer">AutoZI</a> and <a href="https://arxiv.org/pdf/1905.02269.pdf" target="_blank" rel="noopener noreferrer">gimVI</a>) directly into the <code>scvi</code> codebase. Even our most recent work, such as <a href="https://www.nature.com/articles/s41592-020-01050-x" target="_blank" rel="noopener noreferrer">totalVI</a> for CITE-seq modeling, and our <a href="https://papers.nips.cc/paper/2020/file/357a6fdf7642bf815a88822c447d9dc4-Paper.pdf" target="_blank" rel="noopener noreferrer">decision-making procedure</a> for differential expression was also developed this way.</p><p>However, our ambition at the time was to solely use this infrastructure for internal developments. Consequently, some mistakes were made, in which we did not make use of the existing single-cell analysis infrastructure. First, we had built manual data formatting boilerplate code to read many different single-cell omics input formats. We also had incorporated customized plotting code while working on the dataset integration problem. Ideally, all these functions would come from another package, such as <a href="https://scanpy.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer"><code>Scanpy</code></a>. As a second striking example, each algorithm was not straightforward to use as it often required the end user to create multiple objects, such as dataloaders, trainers, and models. This was confusing for the user and also hard to maintain on our end. Indeed, over a year ago we encountered a <a href="https://github.com/YosefLab/scvi-tools/issues/669" target="_blank" rel="noopener noreferrer">bug</a> at <a href="https://github.com/theislab/scib/issues/114" target="_blank" rel="noopener noreferrer">benchmarking time</a>, in which scVI by default was trained on one single-cell (instead of the whole training set)!</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="identifying-key-improvements-creation-of-scvi-tools"></a>Identifying key improvements, creation of <code>scvi-tools</code><a class="hash-link" href="#identifying-key-improvements-creation-of-scvi-tools" title="Direct link to heading">#</a></h2><p>Even though we were actively branding the <code>scvi</code> codebase as a framework for creating new probabilistic models, we did not encounter clear success in this area. A notable exception is the <a href="https://academic.oup.com/bioinformatics/article/36/11/3418/5807606" target="_blank" rel="noopener noreferrer"><code>LDVAE</code> model</a>, which was the first model in <code>scvi</code> developed by an external user of the codebase (thanks <a href="https://www.nxn.se/" target="_blank" rel="noopener noreferrer">Valentine</a>!). Later, we became aware of some suboptimal API choices we had made, and decided to work on improving the user experience in order to make the codebase more attractive. At this point in time, Galen joined the Yosef lab and we rethought the foundations of <code>scvi-tools</code>:</p><ol><li>Host reimplementations of existing methods that are currently difficult to use.</li><li>For all methods, provide a consistent and simplified user experience, and provide <a href="https://docs.scvi-tools.org/en/stable/user_guide/user.html" target="_blank" rel="noopener noreferrer">tutorials</a> that walkthrough a meaningful application.</li><li>Focus on <a href="https://anndata.readthedocs.io/en/latest/" target="_blank" rel="noopener noreferrer"><code>AnnData</code></a> for the input data format (potentially reducing almost half of the code in the codebase), and use <a href="https://scanpy.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer"><code>Scanpy</code></a> for all other processing.</li><li>Build explicit tutorials for interaction with the R ecosystem (<a href="https://www.bioconductor.org/" target="_blank" rel="noopener noreferrer"><code>Bioconductor</code></a>, <a href="https://satijalab.org/seurat/" target="_blank" rel="noopener noreferrer"><code>Seurat</code></a>).</li><li>Rewrite all the training code, relying instead on <a href="https://www.pytorchlightning.ai/" target="_blank" rel="noopener noreferrer">PyTorch Lightning</a>.</li><li>Add an interface to <a href="https://pyro.ai/" target="_blank" rel="noopener noreferrer">Pyro</a>, in order to further automate inference.</li><li>Build <a href="https://docs.scvi-tools.org/en/stable/user_guide/developer.html" target="_blank" rel="noopener noreferrer">tutorials</a> for model developers.</li></ol><p>Today, most of those features are readily usable. Visit our <a href="https://scvi-tools.org" target="_blank" rel="noopener noreferrer">landing page</a>! We detail below some of those developments, present in <code>scvi-tools</code> v0.9.0.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="focus-on-the-model"></a>Focus on the model<a class="hash-link" href="#focus-on-the-model" title="Direct link to heading">#</a></h3><p>In this release, we introduced a new paradigm for building single-cell focused probabilistic models in which model development is hyper-focused on the model at hand. Based on our experience of building variational autoencoders for single-cell data, we identified several opportunties to abstract boilerplate code in a reusable way. Therefore, we built objects in the scvi-tools codebase to handle auxiliary tasks such as data loading, training, save/load, and device management. As an example, we wrote the <code>scvi.data</code> module to handle <code>AnnData</code> objects, including registration of model-specific tensors and generic &quot;ann&quot;-data loading into models. Consequently, model development solely focuses on (1) defining a probabilistic model/inference scheme and (2) expressing it in a structured way based on our abstract classes. Learn more with our <a href="https://docs.scvi-tools.org/en/stable/user_guide/developer.html" target="_blank" rel="noopener noreferrer">tutorials</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="new-implementations"></a>New implementations<a class="hash-link" href="#new-implementations" title="Direct link to heading">#</a></h3><p>With these new model building blocks, we were able to implement models external to our lab with relative ease. This included <a href="https://www.nature.com/articles/s42003-020-01247-y" target="_blank" rel="noopener noreferrer">Stereoscope</a> for deconvolving spatial transcriptomics data, <a href="https://www.sciencedirect.com/science/article/pii/S2405471220301952" target="_blank" rel="noopener noreferrer">Solo</a> for detection of doublets in scRNA-seq data, and <a href="https://www.nature.com/articles/s41592-019-0529-1" target="_blank" rel="noopener noreferrer">CellAssign</a> for reference-based annotation of scRNA-seq data. These implementations required significantly fewer lines of code with <code>scvi-tools</code>.</p><p>For the example of Stereoscope, it took one afternoon and nearly 600 lines of code for its reimplementation in <code>scvi-tools</code> (the original codebase has 6,000 lines of code). Also, the algorithm may now be run directly from AnnData objects in a Jupyter notebook or in Google Colab. This may be more attractive to certain users compared to the original codebase, which was only usable from the command line interface.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="pyro-integration"></a>Pyro integration<a class="hash-link" href="#pyro-integration" title="Direct link to heading">#</a></h3><p>Another feature we are excited about is the integration with <a href="https://pyro.ai/" target="_blank" rel="noopener noreferrer">Pyro</a>, which further abstracts the process of manually deriving optimization objectives. The core Pyro team, who joined the Broad Institute a couple of months ago, recently released a simple <a href="https://pyro.ai/examples/scanvi.html" target="_blank" rel="noopener noreferrer">reimplementation of our scANVI model in Pyro</a>. We therefore highly encourage using Pyro for new model developments, although relying on Pyro to power a model remains completely optional. We anticipate that Pyro will be especially useful for automating inference for complex hierarchical Bayesian models, since writing the automatic differentiation variational inference (ADVI) recipe manually would require many lines of code, and the evidence lower bound would potentially be tedious to write.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="template-repository"></a>Template repository<a class="hash-link" href="#template-repository" title="Direct link to heading">#</a></h3><p>We also wrote a <a href="https://scvi-tools-skeleton.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer">template</a> GitHub repository to accelerate the package creation process. This includes template code to setup documentation, continuous integration testing, and popular code styling practices. Additionally, we incorporated example implementations for scVI in both PyTorch and Pyro.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="new-longitudinal-features"></a>New longitudinal features<a class="hash-link" href="#new-longitudinal-features" title="Direct link to heading">#</a></h3><p>Finally, thanks to our refactoring effort while implementating of all these models in the same codebase, we were able to broadcast new features across models. This included support for multiple (continuous or categorical) covariates when integrating data with scVI, scANVI, or totalVI. We are excited to see the impact of non-linear dataset integration extended in this way, and have already seen promising results in correcting, e.g., cell cycle effects. We also have extended the <a href="https://scarches.readthedocs.io/en/latest/" target="_blank" rel="noopener noreferrer">scArches</a> method for query/reference dataset integration to the scVI, scANVI, and totalVI models. This required implementing one <a href="https://github.com/YosefLab/scvi-tools/blob/master/scvi/model/base/_archesmixin.py" target="_blank" rel="noopener noreferrer"><code>Mixin</code></a> class with the core transfer learning logic.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="next-steps"></a>Next steps<a class="hash-link" href="#next-steps" title="Direct link to heading">#</a></h2><p>We are actively looking for users, as well as feedback! Integration in scvi-tools may be possible in at least two different forms. For example, methods developers may choose to either have their method directly present in the external module of our codebase (such as Stereoscope, gimVI, and CellAssign so far), or clone our template and host their independent repository.</p><p>Please let us know if you would like to add your methods to scvi-tools, or if you have any other suggestions on <a href="https://github.com/YosefLab/scvi-tools/issues" target="_blank" rel="noopener noreferrer">Github issues</a>, <a href="https://discourse.scvi-tools.org/" target="_blank" rel="noopener noreferrer">Discourse</a> or <a href="https://gitter.im/scvi-tools/development" target="_blank" rel="noopener noreferrer">Gitter</a>!</p></section><footer class="row margin-vert--lg"><div class="col"><strong>Tags:</strong><a class="margin-horiz--sm" href="/blog/tags/scvi-tools">scvi-tools</a><a class="margin-horiz--sm" href="/blog/tags/release">release</a></div></footer></article><div><a href="https://github.com/YosefLab/scvi-tools-site/blog/blog/2021-3-03-scvi-tools-v090.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 40 40" class="iconEdit_2LL7"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="margin-vert--xl"><nav class="pagination-nav" aria-label="Blog post page navigation"><div class="pagination-nav__item"></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/blog/autotune"><div class="pagination-nav__sublabel">Older Post</div><div class="pagination-nav__label">Hyperparameter search for scVI »</div></a></div></nav></div></main><div class="col col--2"><div class="tableOfContents_2xL- thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#taking-a-step-back-the-original-scvi-codebase" class="table-of-contents__link">Taking a step back: the original <code>scvi</code> codebase</a></li><li><a href="#identifying-key-improvements-creation-of-scvi-tools" class="table-of-contents__link">Identifying key improvements, creation of <code>scvi-tools</code></a><ul><li><a href="#focus-on-the-model" class="table-of-contents__link">Focus on the model</a></li><li><a href="#new-implementations" class="table-of-contents__link">New implementations</a></li><li><a href="#pyro-integration" class="table-of-contents__link">Pyro integration</a></li><li><a href="#template-repository" class="table-of-contents__link">Template repository</a></li><li><a href="#new-longitudinal-features" class="table-of-contents__link">New longitudinal features</a></li></ul></li><li><a href="#next-steps" class="table-of-contents__link">Next steps</a></li></ul></div></div></div></div></div><footer class="footer footer--dark"><div class="container"><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2021 Yosef Lab, UC Berkeley. Built with Docusaurus.</div></div></div></footer></div>
<script src="/styles.3d86d984.js"></script>
<script src="/runtime~main.de082324.js"></script>
<script src="/main.0990fd90.js"></script>
<script src="/1.28b4df87.js"></script>
<script src="/2.246fa4d6.js"></script>
<script src="/3.5be6f89c.js"></script>
<script src="/ccc49370.122b310d.js"></script>
<script src="/e5436e3b.fe8d3470.js"></script>
<script src="/c534381f.36cfbbd2.js"></script>
</body>
</html>