"use strict";(self.webpackChunkscvi_tools_site=self.webpackChunkscvi_tools_site||[]).push([[965],{3905:function(e,t,a){a.d(t,{Zo:function(){return m},kt:function(){return d}});var n=a(7294);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},i=Object.keys(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var s=n.createContext({}),p=function(e){var t=n.useContext(s),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},m=function(e){var t=p(e.components);return n.createElement(s.Provider,{value:t},e.children)},c={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},u=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,i=e.originalType,s=e.parentName,m=l(e,["components","mdxType","originalType","parentName"]),u=p(a),d=r,f=u["".concat(s,".").concat(d)]||u[d]||c[d]||i;return a?n.createElement(f,o(o({ref:t},m),{},{components:a})):n.createElement(f,o({ref:t},m))}));function d(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=a.length,o=new Array(i);o[0]=u;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l.mdxType="string"==typeof e?e:r,o[1]=l;for(var p=2;p<i;p++)o[p]=a[p];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}u.displayName="MDXCreateElement"},2608:function(e,t,a){a.r(t),a.d(t,{frontMatter:function(){return l},metadata:function(){return s},toc:function(){return p},default:function(){return c}});var n=a(7462),r=a(3366),i=(a(7294),a(3905)),o=["components"],l={},s={type:"mdx",permalink:"/press",source:"@site/src/pages/press.md"},p=[{value:"Articles",id:"articles",children:[]},{value:"Publications",id:"publications",children:[]},{value:"Awards",id:"awards",children:[]},{value:"Podcasts",id:"podcasts",children:[]},{value:"Talks",id:"talks",children:[]},{value:"Blog posts",id:"blog-posts",children:[]}],m={toc:p};function c(e){var t=e.components,a=(0,r.Z)(e,o);return(0,i.kt)("wrapper",(0,n.Z)({},m,a,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"press"},"Press"),(0,i.kt)("h2",{id:"articles"},"Articles"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.embopress.org/doi/full/10.15252/msb.20199198"},(0,i.kt)("em",{parentName:"a"},"Molecular Systems Biology"),": Enhancing scientific discoveries in molecular biology with deep generative models")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.nature.com/articles/s41592-018-0230-9&"},(0,i.kt)("em",{parentName:"a"},"Nature Methods"),": Bayesian deep learning for single-cell analysis")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://engineering.berkeley.edu/news/2021/03/totalvi-a-transformative-algorithm/"},(0,i.kt)("em",{parentName:"a"},"Berkeley Engineering News"),": TotalVI: A transformative algorithm"))),(0,i.kt)("h2",{id:"publications"},"Publications"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Multi-resolution deconvolution of spatial transcriptomics data reveals continuous patterns of inflammation.\nR. Lopez","*",", B. Li","*",", H. Keren-Shaul","*",", P. Boyeau, M. Kedmi, D. Pilzer, A. Jelinski, E. David, A. Wagner, Y. Addadi, M.I. Jordan, I. Amit\u2020, N. Yosef\u2020\n",(0,i.kt)("em",{parentName:"li"},"bioRxiv"),", 2021"),(0,i.kt)("li",{parentName:"ul"},"scvi-tools: a library for deep probabilistic analysis of single-cell omics data.\nA. Gayoso","*",", R. Lopez","*",", G. Xing","*",", P. Boyeau, K. Wu, M. Jayasuriya, E. Mehlman, M. Langevin, Y. Liu, J. Samaran, G. Misrachi, A. Nazaret, O. Clivio, C. Xu, T. Ashuach, M. Lotfollahi, V. Svensson, E. Beltrame, C. Talavera-Lopez, L. Pachter, F.J. Theis, A. Streets, M.I. Jordan, J. Regier, N. Yosef.\n",(0,i.kt)("em",{parentName:"li"},"bioRxiv"),", 2021"),(0,i.kt)("li",{parentName:"ul"},"Joint probabilistic modeling of single-cell multi-omic data with totalVI.\nA. Gayoso","*",", Z. Steier","*",", R. Lopez, J. Regier, KL. Nazor, A. Streets\u2020, N Yosef\u2020.\n",(0,i.kt)("em",{parentName:"li"},"Nature Methods"),", 2021"),(0,i.kt)("li",{parentName:"ul"},"Probabilistic harmonization and annotation of single-cell transcriptomics data with deep generative models.\nC. Xu","*",", R. Lopez","*",", E. Mehlman","*",", J. Regier, M.I. Jordan, N. Yosef.\n",(0,i.kt)("em",{parentName:"li"},"Molecular Systems Biology"),", 2021"),(0,i.kt)("li",{parentName:"ul"},"Decision-making with auto-encoding variational Bayes.\nR. Lopez, P. Boyeau, N. Yosef, M. Jordan, J. Regier.\n",(0,i.kt)("em",{parentName:"li"},"Advances in Neural Information Processing Systems"),", 2020"),(0,i.kt)("li",{parentName:"ul"},"Interpretable factor models of single-cell RNA-seq via variational autoencoders.\nV. Svensson, A. Gayoso, N. Yosef, L. Pachter.\n",(0,i.kt)("em",{parentName:"li"},"Bioinformatics"),", 2020"),(0,i.kt)("li",{parentName:"ul"},"A joint model of unpaired data from scRNA-seq and spatial transcriptomics for imputing missing gene expression measurements.\nR. Lopez","*",", A. Nazaret","*",", M. Langevin","*",", J. Samaran","*",", J. Regier","*",", M.I. Jordan, N. Yosef.\n",(0,i.kt)("em",{parentName:"li"},"ICML Workshop on Computational Biology"),", 2019"),(0,i.kt)("li",{parentName:"ul"},"Deep generative models for detecting differential expression in single cells.\nP. Boyeau, R. Lopez, J. Regier, A. Gayoso, MI. Jordan, N. Yosef.\n",(0,i.kt)("em",{parentName:"li"},"Machine Learning in Computational Biology meeting"),", 2019"),(0,i.kt)("li",{parentName:"ul"},"Detecting zero-inflated denes in single-cell transcriptomics data.\nO. Clivio, R. Lopez, J. Regier, A. Gayoso, MI. Jordan, N Yosef.\n",(0,i.kt)("em",{parentName:"li"},"Machine Learning in Computational Biology meeting"),", 2019"),(0,i.kt)("li",{parentName:"ul"},"Information constraints on auto-encoding variational Bayes.\nR. Lopez, J. Regier, M. Jordan, N. Yosef.\n",(0,i.kt)("em",{parentName:"li"},"Advances in Neural Information Processing Systems"),", 2018"),(0,i.kt)("li",{parentName:"ul"},"A deep generative model for semi-supervised classification with noisy labels.\nM. Langevin, E. Mehlman, J. Regier, R. Lopez, M.I. Jordan, N. Yosef.\n",(0,i.kt)("em",{parentName:"li"},"Bay Area Machine Learning Symposium"),", 2018"),(0,i.kt)("li",{parentName:"ul"},"Deep generative modeling for single-cell transcriptomics.\nR. Lopez, J. Regier, MB. Cole, M. Jordan, N. Yosef.\n",(0,i.kt)("em",{parentName:"li"},"Nature Methods"),", 2018")),(0,i.kt)("h2",{id:"awards"},"Awards"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://facultyopinions.com/prime/734522090"},(0,i.kt)("em",{parentName:"a"},"F1000 exceptional recommendation"),": Deep generative modeling for single-cell transcriptomics")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://sites.google.com/view/icml-compbio-2019/home"},"ICML Workshop on Computational Biology, Best Poster Award, gimVI"))),(0,i.kt)("h2",{id:"podcasts"},"Podcasts"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://bioinformatics.chat/scvi"},(0,i.kt)("em",{parentName:"a"},"The Bioinformatics Chat Podcast"),": scVI with Romain Lopez and Gabriel Misrachi"))),(0,i.kt)("h2",{id:"talks"},"Talks"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://youtu.be/tpf-BgqNxwY?t=5185"},"PyData, AI & Single-Cell Genomics: scVI and scvi-tools")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://youtu.be/NNOkdgE4jNM?t=2783"},(0,i.kt)("em",{parentName:"a"},"MIT 6.874, Deep Learning in the Life"),": Deep generative modeling for single-cell transcriptomics")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.broadinstitute.org/talks/regev-lab-cell-circuits-and-epigenomics-and-mia-special-seminar-deep-generative-modeling"},(0,i.kt)("em",{parentName:"a"},"Regev Lab, Cell Circuits and Epigenomics, and MIA special seminar"),": Deep generative modeling for single-cell transcriptomics")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://ccbskillssem.github.io/pages/scanpy_scvi_tools/"},(0,i.kt)("em",{parentName:"a"},"Berkeley CCB skills seminar"),": Single-cell data analysis with Scanpy and scvi-tools")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.youtube.com/watch?v=A2Sv0AUA27I"},(0,i.kt)("em",{parentName:"a"},"Bay Area Machine Learning Symposium"),": A deep generative model for semi-supervised classification with noisy labels")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.youtube.com/watch?v=NlpqwjQ91Sc"},(0,i.kt)("em",{parentName:"a"},"Bay Area Machine Learning Symposium"),": A deep generative model for gene expression profiles from single-cell RNA sequencing"))),(0,i.kt)("h2",{id:"blog-posts"},"Blog posts"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://bair.berkeley.edu/blog/2018/12/05/genes/"},(0,i.kt)("em",{parentName:"a"},"BAIR Blog"),": Building Gene Expression Atlases with Deep Generative Models for Single-cell Transcriptomics")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.nxn.se/valent/2018/4/20/count-based-autoencoders-and-the-future-for-scrna-seq-analysis"},(0,i.kt)("em",{parentName:"a"},'What do you mean "heterogeneity"?'),": Count based autoencoders and the future for scRNA-seq analysis")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.nxn.se/valent/2018/10/4/integrating-single-cell-and-spatial-frontal-cortex-data-with-scvi"},(0,i.kt)("em",{parentName:"a"},'What do you mean "heterogeneity"?'),": Integrating scRNA-seq and spatial STARmap data from mouse frontal cortex with scVI")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://scvi-tools.org/blog/autotune"},(0,i.kt)("em",{parentName:"a"},"Yosef Lab Blog"),": Hyperparameter search for scVI")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://scvi-tools.org/blog/zero-inflation"},(0,i.kt)("em",{parentName:"a"},"Yosef Lab Blog"),": Should we zero-inflate scVI?")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.nxn.se/valent/2020/9/16/a-trip-through-the-layers-of-scvi"},(0,i.kt)("em",{parentName:"a"},'What do you mean "heterogeneity"?'),": A trip through the layers of scVI")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://saket-choudhary.me/blog/2020/12/27/scvi/"},(0,i.kt)("em",{parentName:"a"},"Saket Choudhary"),": Experiments with scVI"))))}c.isMDXComponent=!0}}]);