"use strict";(self.webpackChunkscvi_tools_site=self.webpackChunkscvi_tools_site||[]).push([[9644],{3905:function(e,t,a){a.d(t,{Zo:function(){return p},kt:function(){return u}});var n=a(7294);function i(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function r(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function s(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?r(Object(a),!0).forEach((function(t){i(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):r(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,i=function(e,t){if(null==e)return{};var a,n,i={},r=Object.keys(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||(i[a]=e[a]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(i[a]=e[a])}return i}var o=n.createContext({}),m=function(e){var t=n.useContext(o),a=t;return e&&(a="function"==typeof e?e(t):s(s({},t),e)),a},p=function(e){var t=m(e.components);return n.createElement(o.Provider,{value:t},e.children)},c={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},d=n.forwardRef((function(e,t){var a=e.components,i=e.mdxType,r=e.originalType,o=e.parentName,p=l(e,["components","mdxType","originalType","parentName"]),d=m(a),u=i,h=d["".concat(o,".").concat(u)]||d[u]||c[u]||r;return a?n.createElement(h,s(s({ref:t},p),{},{components:a})):n.createElement(h,s({ref:t},p))}));function u(e,t){var a=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=a.length,s=new Array(r);s[0]=d;var l={};for(var o in t)hasOwnProperty.call(t,o)&&(l[o]=t[o]);l.originalType=e,l.mdxType="string"==typeof e?e:i,s[1]=l;for(var m=2;m<r;m++)s[m]=a[m];return n.createElement.apply(null,s)}return n.createElement.apply(null,a)}d.displayName="MDXCreateElement"},3919:function(e,t,a){function n(e){return!0===/^(\w*:|\/\/)/.test(e)}function i(e){return void 0!==e&&!n(e)}a.d(t,{b:function(){return n},Z:function(){return i}})},4996:function(e,t,a){a.d(t,{C:function(){return r},Z:function(){return s}});var n=a(2263),i=a(3919);function r(){var e=(0,n.Z)().siteConfig,t=(e=void 0===e?{}:e).baseUrl,a=void 0===t?"/":t,r=e.url;return{withBaseUrl:function(e,t){return function(e,t,a,n){var r=void 0===n?{}:n,s=r.forcePrependBaseUrl,l=void 0!==s&&s,o=r.absolute,m=void 0!==o&&o;if(!a)return a;if(a.startsWith("#"))return a;if((0,i.b)(a))return a;if(l)return t+a;var p=a.startsWith(t)?a:t+a.replace(/^\//,"");return m?e+p:p}(r,a,e,t)}}}function s(e,t){return void 0===t&&(t={}),(0,r().withBaseUrl)(e,t)}},4185:function(e,t,a){a.r(t),a.d(t,{frontMatter:function(){return o},contentTitle:function(){return m},metadata:function(){return p},assets:function(){return c},toc:function(){return d},default:function(){return h}});var n=a(7462),i=a(3366),r=(a(7294),a(3905)),s=a(4996),l=["components"],o={slug:"v13",title:"scvi-tools 1.3 release",date:new Date("2025-07-03T00:00:00.000Z"),author:"Ori Kronfeld, Can Ergen, Nir Yosef",tags:["scvi-tools","release"]},m=void 0,p={permalink:"/blog/v13",editUrl:"https://github.com/YosefLab/scvi-tools-site/blog/blog/2025-3-7-scvi-tools-v1p3.md",source:"@site/blog/2025-3-7-scvi-tools-v1p3.md",title:"scvi-tools 1.3 release",description:"Introduction",date:"2025-07-03T00:00:00.000Z",formattedDate:"July 3, 2025",tags:[{label:"scvi-tools",permalink:"/blog/tags/scvi-tools"},{label:"release",permalink:"/blog/tags/release"}],readingTime:5.715,truncated:!1,authors:[{name:"Ori Kronfeld, Can Ergen, Nir Yosef"}],nextItem:{title:"Mini-batch size in destVI",permalink:"/blog/destvi-batchsize"}},c={authorsImageUrls:[void 0]},d=[{value:"Introduction",id:"introduction",children:[]},{value:"1. \ud83d\udd2c New Models",id:"1--new-models",children:[{value:"<strong>ResolVI</strong>",id:"resolvi",children:[]},{value:"<strong>scVIVA</strong>",id:"scviva",children:[]},{value:"<strong>CytoVI</strong>",id:"cytovi",children:[]},{value:"<strong>VIVS</strong>",id:"vivs",children:[]},{value:"<strong>SysVI</strong>",id:"sysvi",children:[]},{value:"<strong>Decipher</strong>",id:"decipher",children:[]},{value:"<strong>MethylVI</strong>",id:"methylvi",children:[]},{value:"<strong>MethylANVI</strong>",id:"methylanvi",children:[]},{value:"<strong>totalANVI</strong>",id:"totalanvi",children:[]}]},{value:"2. \ud83e\udde9 Custom Dataloaders",id:"2-\ud83e\udde9-custom-dataloaders",children:[]},{value:"3. \u2699\ufe0f Core Enhancements",id:"3-\ufe0f-core-enhancements",children:[{value:"Multi\u2011GPU Training",id:"multigpu-training",children:[]},{value:"scIB\u2011Metrics Optimization",id:"scibmetrics-optimization",children:[]},{value:"Explainability &amp; Interpretability",id:"explainability--interpretability",children:[]}]},{value:"Summary",id:"summary",children:[]},{value:"References",id:"references",children:[]}],u={toc:d};function h(e){var t=e.components,a=(0,i.Z)(e,l);return(0,r.kt)("wrapper",(0,n.Z)({},u,a,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h2",{id:"introduction"},"Introduction"),(0,r.kt)("p",null,"We\u2019re proud to introduce scvi\u2011tools v1.3, encompassing major advances in modeling, data loading, computational scalability, metric integration, and interpretability in single-cell analytics. "),(0,r.kt)("p",null,"Featuring nine new or enhanced models\u2014optimized for spatial, cytometry, methylation, perturbation, and multi\u2011omic data\u2014it also introduces streaming data loaders for large-scale datasets, multi\u2011GPU model training, on-the-fly metric tuning, and integrated model interpretability. "),(0,r.kt)("p",null,"This article delves into each enhancement with depth, including detailed insights, illustrative figures, and manuscript references."),(0,r.kt)("hr",null),(0,r.kt)("h2",{id:"1--new-models"},"1. \ud83d\udd2c New Models"),(0,r.kt)("h3",{id:"resolvi"},(0,r.kt)("strong",{parentName:"h3"},"ResolVI")),(0,r.kt)("p",null,"ResolVI",(0,r.kt)("sup",{parentName:"p",id:"fnref-ref1"},(0,r.kt)("a",{parentName:"sup",href:"#fn-ref1",className:"footnote-ref"},"ref1"))," is a spatial transcriptomics denoising model that reallocates mis-assigned gene counts among true cells, neighborhood leakage, and background. It employs a Gaussian-mixture latent prior to learn corrected counts and interpretable embeddings. This approach is highly scalable (handling >1 million spots) and offers downstream capabilities like differential expression and transfer learning on corrected data"),(0,r.kt)("p",null,"In ",(0,r.kt)("a",{parentName:"p",href:"https://docs.scvi-tools.org/en/stable/tutorials/notebooks/spatial/resolVI_tutorial.html"},"tutorial"),", it has been shown to markedly enhance spatial expression accuracy in noisy segmentation settings, enabling reliable differential expression\u2014especially in high-throughput ST datasets.  "),(0,r.kt)("img",{alt:"ResolVI",width:"100%",src:(0,s.Z)("img/blog-post-scvi-tools-1p3/resolvi.png")}),(0,r.kt)("p",null,"Figure 1: ResolVI cell type annotations based on noisy cellular segmentation Xenium data of a mouse brain. The left hemisphere for model training and the right hemisphere for transfer mapping. Original labeles were based on Leiden clustering of the ProSeg algorithm"),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"scviva"},(0,r.kt)("strong",{parentName:"h3"},"scVIVA")),(0,r.kt)("p",null,"scVIVA",(0,r.kt)("sup",{parentName:"p",id:"fnref-ref2"},(0,r.kt)("a",{parentName:"sup",href:"#fn-ref2",className:"footnote-ref"},"ref2"))," augments spatial transcriptomics analysis by jointly modeling each cell\u2019s own expression and its micro-environmental context (neighborhood composition and gene counts). This niche-aware VAE embeds both cellular identity and environmental features, revealing tissue-specific patterns and environment-driven variation. Its latent embeddings delineate tissue-specific structures\u2014ideal for spatial differential abundance or niche-focused clustering studies."),(0,r.kt)("p",null,"Dedicated ",(0,r.kt)("a",{parentName:"p",href:"https://docs.scvi-tools.org/en/stable/tutorials/notebooks/spatial/scVIVA_tutorial.html"},"tutorial")," showcase how scVIVA enables niche-focused clustering and differential abundance analyses"),(0,r.kt)("img",{alt:"scVIVA",width:"100%",src:(0,s.Z)("img/blog-post-scvi-tools-1p3/scviva.png")}),(0,r.kt)("p",null,"Figure 2: scVIVA results: median Log-Fold Change (LFC) of upregulated genes in ",(0,r.kt)("span",{parentName:"p",className:"math math-inline"},(0,r.kt)("span",{parentName:"span",className:"katex"},(0,r.kt)("span",{parentName:"span",className:"katex-mathml"},(0,r.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,r.kt)("semantics",{parentName:"math"},(0,r.kt)("mrow",{parentName:"semantics"},(0,r.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"G1")),(0,r.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\textit{G1}")))),(0,r.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,r.kt)("span",{parentName:"span",className:"base"},(0,r.kt)("span",{parentName:"span",className:"strut",style:{height:"0.68333em",verticalAlign:"0em"}}),(0,r.kt)("span",{parentName:"span",className:"mord text"},(0,r.kt)("span",{parentName:"span",className:"mord textit"},"G1"))))))," vs ",(0,r.kt)("span",{parentName:"p",className:"math math-inline"},(0,r.kt)("span",{parentName:"span",className:"katex"},(0,r.kt)("span",{parentName:"span",className:"katex-mathml"},(0,r.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,r.kt)("semantics",{parentName:"math"},(0,r.kt)("mrow",{parentName:"semantics"},(0,r.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"G2")),(0,r.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\textit{G2}")))),(0,r.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,r.kt)("span",{parentName:"span",className:"base"},(0,r.kt)("span",{parentName:"span",className:"strut",style:{height:"0.68333em",verticalAlign:"0em"}}),(0,r.kt)("span",{parentName:"span",className:"mord text"},(0,r.kt)("span",{parentName:"span",className:"mord textit"},"G2"))))))," displayed on the x-axis, while we compare differential expression computed between ",(0,r.kt)("span",{parentName:"p",className:"math math-inline"},(0,r.kt)("span",{parentName:"span",className:"katex"},(0,r.kt)("span",{parentName:"span",className:"katex-mathml"},(0,r.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,r.kt)("semantics",{parentName:"math"},(0,r.kt)("mrow",{parentName:"semantics"},(0,r.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"N1")),(0,r.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\textit{N1}")))),(0,r.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,r.kt)("span",{parentName:"span",className:"base"},(0,r.kt)("span",{parentName:"span",className:"strut",style:{height:"0.68333em",verticalAlign:"0em"}}),(0,r.kt)("span",{parentName:"span",className:"mord text"},(0,r.kt)("span",{parentName:"span",className:"mord textit"},"N1")))))),"  and ",(0,r.kt)("span",{parentName:"p",className:"math math-inline"},(0,r.kt)("span",{parentName:"span",className:"katex"},(0,r.kt)("span",{parentName:"span",className:"katex-mathml"},(0,r.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,r.kt)("semantics",{parentName:"math"},(0,r.kt)("mrow",{parentName:"semantics"},(0,r.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"G2")),(0,r.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\textit{G2}")))),(0,r.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,r.kt)("span",{parentName:"span",className:"base"},(0,r.kt)("span",{parentName:"span",className:"strut",style:{height:"0.68333em",verticalAlign:"0em"}}),(0,r.kt)("span",{parentName:"span",className:"mord text"},(0,r.kt)("span",{parentName:"span",className:"mord textit"},"G2"))))))," on the y-axis.\nGenes are colored by their marker label (yellow=significantly upregulated in ",(0,r.kt)("span",{parentName:"p",className:"math math-inline"},(0,r.kt)("span",{parentName:"span",className:"katex"},(0,r.kt)("span",{parentName:"span",className:"katex-mathml"},(0,r.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,r.kt)("semantics",{parentName:"math"},(0,r.kt)("mrow",{parentName:"semantics"},(0,r.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"G1")),(0,r.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\textit{G1}")))),(0,r.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,r.kt)("span",{parentName:"span",className:"base"},(0,r.kt)("span",{parentName:"span",className:"strut",style:{height:"0.68333em",verticalAlign:"0em"}}),(0,r.kt)("span",{parentName:"span",className:"mord text"},(0,r.kt)("span",{parentName:"span",className:"mord textit"},"G1"))))))," vs ",(0,r.kt)("span",{parentName:"p",className:"math math-inline"},(0,r.kt)("span",{parentName:"span",className:"katex"},(0,r.kt)("span",{parentName:"span",className:"katex-mathml"},(0,r.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,r.kt)("semantics",{parentName:"math"},(0,r.kt)("mrow",{parentName:"semantics"},(0,r.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"N1")),(0,r.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\textit{N1}")))),(0,r.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,r.kt)("span",{parentName:"span",className:"base"},(0,r.kt)("span",{parentName:"span",className:"strut",style:{height:"0.68333em",verticalAlign:"0em"}}),(0,r.kt)("span",{parentName:"span",className:"mord text"},(0,r.kt)("span",{parentName:"span",className:"mord textit"},"N1")))))),", green otherwise).\nWe also display the classifier decision boundary (the predicted probability of being in the yellow class). "),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"cytovi"},(0,r.kt)("strong",{parentName:"h3"},"CytoVI")),(0,r.kt)("p",null,"CytoVI brings totalVI-inspired modeling to cytometry and mass cytometry data. It models protein-marker distributions, corrects for dropouts and technical batch variation, and generates embeddings for downstream clustering and abundance inference. Tutorials demonstrate CytoVI\u2019s capability to distinguish subtle immune subpopulations across batches, bringing probabilistic rigor to CyTOF analysis."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"vivs"},(0,r.kt)("strong",{parentName:"h3"},"VIVS")),(0,r.kt)("p",null,"Variational Inference for Variable Selection (VIVS) performs interpretable cross-modality feature selection\u2014like gene\u2013protein associations\u2014while formally controlling false discoveries. Illustrated in ",(0,r.kt)("em",{parentName:"p"},"Genome Biology")," 2024, VIVS effectively maps regulatory relationships in multi-omic datasets, balancing interpretability and scalability with conditional randomization testing and latent selection."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"sysvi"},(0,r.kt)("strong",{parentName:"h3"},"SysVI")),(0,r.kt)("p",null,"SysVI tackles major batch effects\u2014such as those arising from cross-species or organoid-versus-tissue studies\u2014using latent cycle-consistency and VampPrior regularization. Compared to Harmony or scVI, SysVI excels at aligning technical systems while preserving true biological variance, producing embedding spaces where analogous cell types across batches cluster coherently."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"decipher"},(0,r.kt)("strong",{parentName:"h3"},"Decipher")),(0,r.kt)("p",null,"Designed to dissect perturbation effects (e.g., disease versus control), Decipher disentangles shared and condition-specific variation within a VAE framework. Demonstrated on AML data, it uncovers latent axes aligning with known disease signatures and identifies corresponding differentially expressed genes\u2014bridging latent space analysis and functional biology."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"methylvi"},(0,r.kt)("strong",{parentName:"h3"},"MethylVI")),(0,r.kt)("p",null,"MethylVI is a VAE tailored for single-cell bisulfite sequencing (scBS\u2011seq). By modeling methylation probabilities at genomic regions, it captures epigenetic heterogeneity and learns latent spaces that integrate multiple batches. Tutorials show it outperforms linear methods like PCA in retaining biologically meaningful methylation structures."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"methylanvi"},(0,r.kt)("strong",{parentName:"h3"},"MethylANVI")),(0,r.kt)("p",null,"MethylANVI extends the MethylVI framework with annotation-aware modeling: it jointly integrates methylation profiles with metadata-driven cell-type labels. This supervised model supports both clustering and label transfer, all while capturing latent biological variation across methylome profiles\u2014ideal for epigenetic atlas-building."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"totalanvi"},(0,r.kt)("strong",{parentName:"h3"},"totalANVI")),(0,r.kt)("p",null,"totalANVI brings supervised annotation to CITE\u2011seq-style multi-omic integration. Leveraging a VAE for RNA + protein, it jointly learns latent embeddings and cell-type classifiers. The model simultaneously performs dimensionality reduction, denoising, differential expression, and accurate cell-type annotation in one cohesive model."),(0,r.kt)("hr",null),(0,r.kt)("h2",{id:"2-\ud83e\udde9-custom-dataloaders"},"2. \ud83e\udde9 Custom Dataloaders"),(0,r.kt)("p",null,"scvi\u2011tools v1.3 introduces three scalable streaming dataloaders\u2014LaminDB, Census, and AnnCollection\u2014enabling out-of-core and federated training without memory overload."),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"LaminDB")," streams data directly from Lamindb collections, letting users train SCVI-like models on terabyte-scale datasets stored on disk, with no changes to the training pipeline or performance drop-offs.")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"Census")," employs TileDB for atlas-scale tensor-backed data, offering similar streaming capabilities but enhanced support for multi-dimensional genomic inputs and federated study designs.")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"AnnCollection")," allows seamless union of multiple AnnData sources\u2014perfect for interfacing independent experiments\u2014without manual concatenation or feature alignment. The system handles missing modalities and automatically harmonizes variables across datasets."))),(0,r.kt)("p",null,"These backends support full compatibility with scvi\u2011tools' data registration and training workflows, offering both scale and convenience to large projects."),(0,r.kt)("hr",null),(0,r.kt)("h2",{id:"3-\ufe0f-core-enhancements"},"3. \u2699\ufe0f Core Enhancements"),(0,r.kt)("h3",{id:"multigpu-training"},"Multi\u2011GPU Training"),(0,r.kt)("p",null,"Built on PyTorch Lightning, v1.3 empowers all major models (SCVI, totalVI, SysVI, etc.) to run across multiple GPUs with a single API flag. Training benchmarks on million-cell datasets show training times reduced by 2\u20133\xd7, with full gradient synchronization and no code modifications needed."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"scibmetrics-optimization"},"scIB\u2011Metrics Optimization"),(0,r.kt)("p",null,"scvi\u2011tools now integrates evaluation of scIB clustering metrics (like NMI and batch ARI) during training via ",(0,r.kt)("inlineCode",{parentName:"p"},"ScibCallback"),", and supports hyperparameter tuning via ",(0,r.kt)("inlineCode",{parentName:"p"},"AutotuneExperiment"),". This enables scientists to automatically optimize latent dimension size, learning rate, or architecture based on validation performance\u2014making the model training process more principled and outcome-driven."),(0,r.kt)("hr",null),(0,r.kt)("h3",{id:"explainability--interpretability"},"Explainability & Interpretability"),(0,r.kt)("p",null,"v1.3 brings native support for ",(0,r.kt)("strong",{parentName:"p"},"Integrated Gradients")," (IG) across generative models like CondSCVI, peakVI, and totalVI. Users can compute gene-level attribution scores tied to latent dimensions or differential axes. This complements ",(0,r.kt)("inlineCode",{parentName:"p"},"get_normalized_expression"),", delivering a full pipeline that links model representations back to biologically interpretable molecular mechanisms."),(0,r.kt)("hr",null),(0,r.kt)("h2",{id:"summary"},"Summary"),(0,r.kt)("p",null,"scvi\u2011tools v1.3 is a landmark release that advances the field across three foundational pillars:"),(0,r.kt)("ol",null,(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"Innovative modeling")," across nine tailored VAEs for spatial, protein, methylation, perturbation, and multi\u2011omic data.")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"Scalable data processing")," with three new streaming backends, enabling efficient handling of federated, atlas-scale datasets.")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"Infrastructure and transparency")," with multi-GPU training, metric-aware tuning, and demonstrable model interpretability."))),(0,r.kt)("p",null,"Together, these developments empower researchers to build, train, and interpret probabilistic models at scale\u2014in a reproducible, transparent, and biologically meaningful way."),(0,r.kt)("hr",null),(0,r.kt)("h2",{id:"references"},"References"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"  ResolVI: addressing noise and bias in spatial transcriptomics / Ergen et al.")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"  scVIVA: a probabilistic framework for representation of cells and their environments in spatial transcriptomics / Levy et al.")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cA unified method for differential methylation analysis\u201d \u2013 first author: Lopez  ")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cJoint probabilistic modeling of single-cell multi-omic data with totalVI\u201d \u2013 Gayoso  ")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cIntegrating single-cell RNA-seq datasets with substantial batch effects\u201d \u2013 Hrovatin  ")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cJoint representation and visualization of derailed cell states with Decipher\u201d \u2013 Nazaret  ")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cVariational Inference for Variable Selection\u201d \u2013 (VIVS)  ")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cMethylVI: a generative model of scBS-seq data\u201d \u2013 (single author missing)  ")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cResolVI: probabilistic deconvolution of spatial transcriptomics\u201d \u2013 (if available)  ")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"\u201cscVIVA: niche-aware spatial VAE\u201d \u2013 (if available)"))),(0,r.kt)("div",{className:"footnotes"},(0,r.kt)("hr",{parentName:"div"}),(0,r.kt)("ol",{parentName:"div"},(0,r.kt)("li",{parentName:"ol",id:"fn-ref1"},(0,r.kt)("a",{parentName:"li",href:"#fnref-ref1",className:"footnote-backref"},"\u21a9")),(0,r.kt)("li",{parentName:"ol",id:"fn-ref2"},(0,r.kt)("a",{parentName:"li",href:"#fnref-ref2",className:"footnote-backref"},"\u21a9")))))}h.isMDXComponent=!0}}]);