<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.70">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="scvi-tools Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="scvi-tools Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-141905405-3","auto"),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css"><title data-react-helmet="true">scvi-tools</title><meta data-react-helmet="true" property="og:title" content="scvi-tools"><meta data-react-helmet="true" property="og:url" content="https://scvi-tools.org/get_started"><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_tag" content="default"><link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://scvi-tools.org/get_started"><link rel="stylesheet" href="/styles.4e7412cc.css">
<link rel="stylesheet" href="/main.b90caea9.css">
<link rel="preload" href="/styles.3d86d984.js" as="script">
<link rel="preload" href="/runtime~main.c3018b38.js" as="script">
<link rel="preload" href="/main.d238dfa4.js" as="script">
<link rel="preload" href="/1.28b4df87.js" as="script">
<link rel="preload" href="/2.246fa4d6.js" as="script">
<link rel="preload" href="/39.2e65902b.js" as="script">
<link rel="preload" href="/b47ec908.0cf34ec8.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_11B0">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">scvi-tools</strong></a></div><div class="navbar__items navbar__items--right"><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/get_started">Get Started</a><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Docs</a><ul class="dropdown__menu"><li><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" class="dropdown__link">Full documentation</a></li><li><a href="https://docs.scvi-tools.org/en/stable/user_guide/user.html" target="_blank" rel="noopener noreferrer" class="dropdown__link">User guide</a></li><li><a href="https://docs.scvi-tools.org/en/stable/user_guide/developer.html" target="_blank" rel="noopener noreferrer" class="dropdown__link">Developer guide</a></li><li><a href="https://docs.scvi-tools.org/en/stable/api/index.html" target="_blank" rel="noopener noreferrer" class="dropdown__link">API reference</a></li></ul></div><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a class="navbar__item navbar__link">About</a><ul class="dropdown__menu"><li><a class="dropdown__link" href="/team">Team</a></li><li><a class="dropdown__link" href="/news">News</a></li><li><a class="dropdown__link" href="/press">Press</a></li><li><a class="dropdown__link" href="/ecosystem">Ecosystem</a></li></ul></div><a class="navbar__item navbar__link" href="/blog">Blog</a><a href="https://discourse.scvi-tools.org/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Discussion</a><a href="https://github.com/YosefLab/scvi-tools" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub</a></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">scvi-tools</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/get_started">Get Started</a></li><li class="menu__list-item menu__list-item--collapsed"><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" role="button" class="menu__link menu__link--sublist">Docs</a><ul class="menu__list"><li class="menu__list-item"><a href="https://docs.scvi-tools.org" target="_blank" rel="noopener noreferrer" class="menu__link">Full documentation</a></li><li class="menu__list-item"><a href="https://docs.scvi-tools.org/en/stable/user_guide/user.html" target="_blank" rel="noopener noreferrer" class="menu__link">User guide</a></li><li class="menu__list-item"><a href="https://docs.scvi-tools.org/en/stable/user_guide/developer.html" target="_blank" rel="noopener noreferrer" class="menu__link">Developer guide</a></li><li class="menu__list-item"><a href="https://docs.scvi-tools.org/en/stable/api/index.html" target="_blank" rel="noopener noreferrer" class="menu__link">API reference</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a role="button" class="menu__link menu__link--sublist">About</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/team">Team</a></li><li class="menu__list-item"><a class="menu__link" href="/news">News</a></li><li class="menu__list-item"><a class="menu__link" href="/press">Press</a></li><li class="menu__list-item"><a class="menu__link" href="/ecosystem">Ecosystem</a></li></ul></li><li class="menu__list-item"><a class="menu__link" href="/blog">Blog</a></li><li class="menu__list-item"><a href="https://discourse.scvi-tools.org/" target="_blank" rel="noopener noreferrer" class="menu__link">Discussion</a></li><li class="menu__list-item"><a href="https://github.com/YosefLab/scvi-tools" target="_blank" rel="noopener noreferrer" class="menu__link">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper"><main><div class="container container--fluid"><div class="margin-vert--lg padding-vert--lg"><div class="row"><div class="col col--8 col--offset-2"><div class="container"><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="get-started"></a>Get Started<a class="hash-link" href="#get-started" title="Direct link to heading">#</a></h1><p><code>scvi-tools</code> (single-cell variational inference tools) is a package for probabilistic modeling of single-cell omics data, built on top of <a href="https://pytorch.org/" target="_blank" rel="noopener noreferrer">PyTorch</a> and <a href="https://anndata.readthedocs.io/en/latest/" target="_blank" rel="noopener noreferrer">AnnData</a>. The package hosts implementations of several models that perform a wide range of single-cell data analysis tasks, as well as the building blocks to rapidly prototype new probabilistic models.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="basic-installation"></a>Basic installation<a class="hash-link" href="#basic-installation" title="Direct link to heading">#</a></h2><p><code>scvi-tools</code> can be installed with <code>pip</code> or <code>conda</code>:</p><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-python codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">pip install scvi</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">tools</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-python codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#393A34;background-color:#f6f8fa"><div class="token-line" style="color:#393A34"><span class="token plain">conda install scvi</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">tools </span><span class="token operator" style="color:#393A34">-</span><span class="token plain">c bioconda </span><span class="token operator" style="color:#393A34">-</span><span class="token plain">c conda</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">forge</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><p>To leverage a GPU for inference, a version of <a href="https://pytorch.org/get-started/locally/" target="_blank" rel="noopener noreferrer">PyTorch</a> that supports your GPU will need to be installed separately.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="single-cell-omics-data-analysis"></a>Single-cell omics data analysis<a class="hash-link" href="#single-cell-omics-data-analysis" title="Direct link to heading">#</a></h2><p><code>scvi-tools</code> hosts implementations of many models, with a convenient interface to <a href="https://scanpy.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer">scanpy</a>, <a href="https://satijalab.org/seurat/" target="_blank" rel="noopener noreferrer">Seurat</a> and <a href="https://www.bioconductor.org/" target="_blank" rel="noopener noreferrer">Bioconductor</a> (refer to our tutorials to learn about our R support):</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="scrna-seq"></a>scRNA-seq<a class="hash-link" href="#scrna-seq" title="Direct link to heading">#</a></h3><ul><li><a href="https://rdcu.be/bdHYQ" target="_blank" rel="noopener noreferrer">scVI</a> for analysis of single-cell RNA-seq data, as well as its improved <a href="https://www.biorxiv.org/content/biorxiv/early/2019/10/04/794289.full.pdf" target="_blank" rel="noopener noreferrer">differential expression framework</a>.</li><li><a href="https://academic.oup.com/bioinformatics/article/36/11/3418/5807606" target="_blank" rel="noopener noreferrer">LDVAE</a> for an interpretable linear factor model version of scVI.</li><li><a href="https://www.embopress.org/doi/full/10.15252/msb.20209620" target="_blank" rel="noopener noreferrer">scANVI</a> for cell annotation of scRNA-seq data using semi-labeled examples.</li><li><a href="https://www.nature.com/articles/s41592-019-0529-1" target="_blank" rel="noopener noreferrer">CellAssign</a> for cell annotation of scRNA-seq data using marker genes. </li><li><a href="https://www.sciencedirect.com/science/article/pii/S2405471220301952" target="_blank" rel="noopener noreferrer">SOLO</a> for detection of doublets.</li><li><a href="https://www.biorxiv.org/content/10.1101/2020.07.16.205997v1" target="_blank" rel="noopener noreferrer">scArches</a> for transfer learning when learning cell embeddings in an online fashion (implemented for scVI, scANVI &amp; totalVI)</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="cite-seq"></a>CITE-seq<a class="hash-link" href="#cite-seq" title="Direct link to heading">#</a></h3><ul><li><a href="https://www.nature.com/articles/s41592-020-01050-x.epdf?sharing_token=F_xpBUwgD5vHVAB5jvU0c9RgN0jAjWel9jnR3ZoTv0NDA6nPyssLQkWDcY8VB7gLKdlD8q3cqpF5ovLZjZTudaweTnzODEXCrkjJ2UJEEpOAhfOCwEZ0h0Y2WTkn_GOBfGUZhH-lzX_u_AswArGFGinU1pcuXPvuTkPrrQI4zko%3D" target="_blank" rel="noopener noreferrer">totalVI</a> for analysis of CITE-seq data.</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="spatial-transcriptomics"></a>Spatial transcriptomics<a class="hash-link" href="#spatial-transcriptomics" title="Direct link to heading">#</a></h3><ul><li><a href="https://www.nature.com/articles/s42003-020-01247-y" target="_blank" rel="noopener noreferrer">Steroscope</a> for deconvolution of spatial transcriptomics data.</li><li><a href="https://arxiv.org/pdf/1905.02269.pdf" target="_blank" rel="noopener noreferrer">gimVI</a> for imputation of missing genes in spatial transcriptomics from scRNA-seq data.</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="scatac-seq"></a>scATAC-seq<a class="hash-link" href="#scatac-seq" title="Direct link to heading">#</a></h3><ul><li>peakVI for analysis of single-cell ATAC sequencing profiles</li></ul><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="methods-developer-toolkit"></a>Methods developer toolkit<a class="hash-link" href="#methods-developer-toolkit" title="Direct link to heading">#</a></h2><p><code>scvi-tools</code> is a high-level probabilistic programming package especially developped for single-cell omics data. We hope it will empower method developers and help them focus on building fruitful models. To achieve this, we host black-box primitives for loading data from <a href="https://anndata.readthedocs.io/en/latest/" target="_blank" rel="noopener noreferrer">AnnData</a>, black-box training procedures using <a href="https://www.pytorchlightning.ai/" target="_blank" rel="noopener noreferrer">PyTorch Lightning</a>, and optionally provides automatic inference recipes based on <a href="https://pyro.ai/" target="_blank" rel="noopener noreferrer">Pyro</a>. We currently support many optimization-based inference procedures including SVI, AEVB, MAP, MLE and EM. To learn more about the developer toolkit, please visit our <a href="https://docs.scvi-tools.org/en/latest/user_guide/index.html" target="_blank" rel="noopener noreferrer">tutorials</a>. We also provide an example <a href="https://github.com/YosefLab/scvi-tools-skeleton/" target="_blank" rel="noopener noreferrer">skeleton</a> implementation of a basic scVI in PyTorch and Pyro as an independant package based off <code>scvi-tools</code>. If you wish to add an implementation of your algorithm in scvi-tools, please post a <a href="https://github.com/YosefLab/scvi-tools/issues" target="_blank" rel="noopener noreferrer">GitHub issue</a> to reach out to us.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="resources"></a>Resources<a class="hash-link" href="#resources" title="Direct link to heading">#</a></h2><ul><li>Tutorials, API reference, and advanced installation guides are available in the <a href="https://docs.scvi-tools.org/en/stable/" target="_blank" rel="noopener noreferrer">docs</a>.</li><li>For discussion of usage, checkout out our <a href="https://discourse.scvi-tools.org/" target="_blank" rel="noopener noreferrer">forum</a>.</li><li>For immediate public chat with the developers, come talk to us on <a href="https://gitter.im/scvi-tools/" target="_blank" rel="noopener noreferrer">Gitter</a>.</li><li>For a skeleton of implementation of a new model in scvi-tools, refer to our <a href="https://github.com/YosefLab/scvi-tools-skeleton/" target="_blank" rel="noopener noreferrer">template repository</a></li><li>To report bugs, make an issue on <a href="https://github.com/YosefLab/scvi-tools/issues" target="_blank" rel="noopener noreferrer">GitHub</a>.</li></ul></div></div><div class="col col--2"><div class="tableOfContents_2xL- thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#basic-installation" class="table-of-contents__link">Basic installation</a></li><li><a href="#single-cell-omics-data-analysis" class="table-of-contents__link">Single-cell omics data analysis</a><ul><li><a href="#scrna-seq" class="table-of-contents__link">scRNA-seq</a></li><li><a href="#cite-seq" class="table-of-contents__link">CITE-seq</a></li><li><a href="#spatial-transcriptomics" class="table-of-contents__link">Spatial transcriptomics</a></li><li><a href="#scatac-seq" class="table-of-contents__link">scATAC-seq</a></li></ul></li><li><a href="#methods-developer-toolkit" class="table-of-contents__link">Methods developer toolkit</a></li><li><a href="#resources" class="table-of-contents__link">Resources</a></li></ul></div></div></div></div></div></main></div><footer class="footer footer--dark"><div class="container"><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2021 Yosef Lab, UC Berkeley. Built with Docusaurus.</div></div></div></footer></div>
<script src="/styles.3d86d984.js"></script>
<script src="/runtime~main.c3018b38.js"></script>
<script src="/main.d238dfa4.js"></script>
<script src="/1.28b4df87.js"></script>
<script src="/2.246fa4d6.js"></script>
<script src="/39.2e65902b.js"></script>
<script src="/b47ec908.0cf34ec8.js"></script>
</body>
</html>