<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.6">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="scvi-tools Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="scvi-tools Blog Atom Feed">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.12.0/dist/katex.min.css"><title data-react-helmet="true">2 posts tagged with &quot;scvi-tools&quot; | scvi-tools</title><meta data-react-helmet="true" property="og:title" content="2 posts tagged with &quot;scvi-tools&quot; | scvi-tools"><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://scvi-tools.org/blog/tags/scvi-tools"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_tag" content="blog_tags_posts"><link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://scvi-tools.org/blog/tags/scvi-tools"><link data-react-helmet="true" rel="alternate" href="https://scvi-tools.org/blog/tags/scvi-tools" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://scvi-tools.org/blog/tags/scvi-tools" hreflang="x-default"><link rel="stylesheet" href="/assets/css/styles.34d598f1.css">
<link rel="preload" href="/assets/js/runtime~main.25fa66c4.js" as="script">
<link rel="preload" href="/assets/js/main.5868d4e3.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_1VuW themedImage--light_3UqQ navbar__logo"><img src="/img/logo.svg" alt="scvi-tools Logo" class="themedImage_1VuW themedImage--dark_hz6m navbar__logo"><b class="navbar__title">scvi-tools</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link" href="/get_started">Get Started</a><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a class="navbar__link">Docs</a><ul class="dropdown__menu"><li><a href="https://docs.scvi-tools.org" target="_self" rel="noopener noreferrer" class="dropdown__link"><span>Full documentation<svg width="12" height="12" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li><li><a href="https://docs.scvi-tools.org/en/stable/tutorials/index.html" target="_self" rel="noopener noreferrer" class="dropdown__link"><span>Tutorials<svg width="12" height="12" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li><li><a href="https://docs.scvi-tools.org/en/stable/user_guide/index.html" target="_self" rel="noopener noreferrer" class="dropdown__link"><span>User guide<svg width="12" height="12" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li><li><a href="https://docs.scvi-tools.org/en/stable/api/index.html" target="_self" rel="noopener noreferrer" class="dropdown__link"><span>API reference<svg width="12" height="12" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a class="navbar__link">About</a><ul class="dropdown__menu"><li><a class="dropdown__link" href="/team">Team</a></li><li><a class="dropdown__link" href="/press">Press</a></li><li><a class="dropdown__link" href="/ecosystem">Ecosystem</a></li></ul></div><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/blog">Blog</a><a href="https://discourse.scvi-tools.org/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link"><span>Discussion<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a><a href="https://github.com/YosefLab/scvi-tools" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper blog-wrapper blog-tags-post-list-page"><div class="container margin-vert--lg"><div class="row"><aside class="col col--3"><nav class="sidebar_2ahu thin-scrollbar" aria-label="Blog recent posts navigation"><div class="sidebarItemTitle_2hhb margin-bottom--md">Recent posts</div><ul class="sidebarItemList_2xAf"><li class="sidebarItem_2UVv"><a class="sidebarItemLink_1RT6" href="/blog/v13">scvi-tools 1.4 release</a></li><li class="sidebarItem_2UVv"><a class="sidebarItemLink_1RT6" href="/blog/destvi-batchsize">Mini-batch size in destVI</a></li><li class="sidebarItem_2UVv"><a class="sidebarItemLink_1RT6" href="/blog/v090">scvi-tools 0.9.0 release</a></li><li class="sidebarItem_2UVv"><a class="sidebarItemLink_1RT6" href="/blog/autotune">Hyperparameter search for scVI</a></li><li class="sidebarItem_2UVv"><a class="sidebarItemLink_1RT6" href="/blog/zero-inflation">Should we zero-inflate scVI?</a></li></ul></nav></aside><main class="col col--7" itemscope="" itemtype="http://schema.org/Blog"><header class="margin-bottom--xl"><h1>2 posts tagged with &quot;scvi-tools&quot;</h1><a href="/blog/tags">View All Tags</a></header><article class="margin-bottom--xl" itemprop="blogPost" itemscope="" itemtype="http://schema.org/BlogPosting"><header><h2 class="blogPostTitle_GeHD" itemprop="headline"><a itemprop="url" href="/blog/v13">scvi-tools 1.4 release</a></h2><div class="blogPostData_291c margin-vert--md"><time datetime="2025-07-03T00:00:00.000Z" itemprop="datePublished">July 3, 2025</time> · 9 min read</div><div class="row margin-top--md margin-bottom--sm"><div class="col col--6 authorCol_1R69"><div class="avatar margin-bottom--sm"><div class="avatar__intro" itemprop="author" itemscope="" itemtype="https://schema.org/Person"><div class="avatar__name"><a itemprop="url"><span itemprop="name">Ori Kronfeld</span></a></div></div></div></div></div></header><div class="markdown" itemprop="articleBody"><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction"></a>Introduction<a class="hash-link" href="#introduction" title="Direct link to heading">#</a></h2><p>We’re proud to introduce scvi‑tools v1.4, encompassing major advances in modeling, data loading, computational scalability, metric integration, and interpretability in single-cell analytics. </p><p>Featuring nine new or enhanced models, optimized for spatial, cytometry, methylation, perturbation, and multi‑omic data, it also introduces custom data loaders for large-scale datasets, multi‑GPU model training, on-the-fly metric tuning, and integrated model interpretability. </p><p>This article delves into each enhancement with depth, including detailed insights, illustrative figures, and manuscript references.</p><hr><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="1--new-models"></a>1. 🔬 New Models<a class="hash-link" href="#1--new-models" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="resolvi"></a>ResolVI<a class="hash-link" href="#resolvi" title="Direct link to heading">#</a></h3><p>ResolVI<sup id="fnref-1"><a href="#fn-1" class="footnote-ref">1</a></sup> is a spatial transcriptomics denoising model that reallocates mis-assigned gene counts among true cells, neighborhood leakage, and background. It employs a Gaussian-mixture latent prior to learn corrected counts and interpretable embeddings. This approach is highly scalable (handling &gt;1 million spots) and offers downstream capabilities like differential expression and transfer learning on corrected data</p><p>In <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/spatial/resolVI_tutorial.html" target="_blank" rel="noopener noreferrer">tutorial</a>, it has been shown to markedly enhance spatial expression accuracy in noisy segmentation settings, enabling reliable differential expression especially in high-throughput ST datasets.  </p><img alt="ResolVI" width="100%" src="/img/blog-post-scvi-tools-1p3/resolvi.png"><p>Figure 1: ResolVI cell type annotations based on noisy cellular segmentation Xenium data of a mouse brain. The left hemisphere for model training and the right hemisphere for transfer mapping. </p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="scviva"></a>scVIVA<a class="hash-link" href="#scviva" title="Direct link to heading">#</a></h3><p>scVIVA<sup id="fnref-2"><a href="#fn-2" class="footnote-ref">2</a></sup> augments spatial transcriptomics analysis by jointly modeling each cell’s own expression and its micro-environmental context (neighborhood composition and gene counts). This niche-aware VAE embeds both cellular identity and environmental features, revealing tissue-specific patterns and environment-driven variation. Its latent embeddings delineate tissue-specific structures - ideal for spatial differential abundance or niche-focused clustering studies.</p><p>Dedicated <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/spatial/scVIVA_tutorial.html" target="_blank" rel="noopener noreferrer">tutorial</a> showcase how scVIVA enables niche-focused clustering and differential abundance analyses</p><img alt="scVIVA" width="100%" src="/img/blog-post-scvi-tools-1p3/scviva.png"><p>Figure 2: scVIVA results: median Log-Fold Change (LFC) of upregulated genes in <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="italic">G1</mtext></mrow><annotation encoding="application/x-tex">\textit{G1}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em"></span><span class="mord text"><span class="mord textit">G1</span></span></span></span></span></span> vs <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="italic">G2</mtext></mrow><annotation encoding="application/x-tex">\textit{G2}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em"></span><span class="mord text"><span class="mord textit">G2</span></span></span></span></span></span> displayed on the x-axis, while we compare differential expression computed between <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="italic">N1</mtext></mrow><annotation encoding="application/x-tex">\textit{N1}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em"></span><span class="mord text"><span class="mord textit">N1</span></span></span></span></span></span>  and <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="italic">G2</mtext></mrow><annotation encoding="application/x-tex">\textit{G2}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em"></span><span class="mord text"><span class="mord textit">G2</span></span></span></span></span></span> on the y-axis.
Genes are colored by their marker label (yellow=significantly upregulated in <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="italic">G1</mtext></mrow><annotation encoding="application/x-tex">\textit{G1}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em"></span><span class="mord text"><span class="mord textit">G1</span></span></span></span></span></span> vs <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="italic">N1</mtext></mrow><annotation encoding="application/x-tex">\textit{N1}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em"></span><span class="mord text"><span class="mord textit">N1</span></span></span></span></span></span>, green otherwise).
We also display the classifier decision boundary (the predicted probability of being in the yellow class). </p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="cytovi"></a>CytoVI<a class="hash-link" href="#cytovi" title="Direct link to heading">#</a></h3><p>CytoVI<sup id="fnref-3"><a href="#fn-3" class="footnote-ref">3</a></sup> brings totalVI-inspired modeling to cytometry and mass cytometry data. It models protein-marker distributions, corrects for dropouts and technical batch variation, and generates embeddings for downstream clustering and abundance inference. </p><p>Early <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/cytometry/CytoVI_batch_correction_tutorial.html" target="_blank" rel="noopener noreferrer">tutorial</a> already demonstrate clear delineation of immune subpopulations across batch-affected datasets.</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="vivs"></a>VIVS<a class="hash-link" href="#vivs" title="Direct link to heading">#</a></h3><p>Variational Inference for Variable Selection (VIVS<sup id="fnref-4"><a href="#fn-4" class="footnote-ref">4</a></sup>) identifies associations across modalities such as gene–protein couplings—while rigorously controlling false discovery rates using conditional randomization. VIVS achieves interpretable and scalable feature selection, enabling discovery of biologically meaningful links in paired datasets</p><p>Tutorial in scvi-tools soon to be updated.</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="sysvi"></a>SysVI<a class="hash-link" href="#sysvi" title="Direct link to heading">#</a></h3><p>SysVI<sup id="fnref-5"><a href="#fn-5" class="footnote-ref">5</a></sup> tackles major batch effects, such as those arising from cross-species or organoid-versus-tissue studies—using latent cycle-consistency and VampPrior regularization. Compared to Harmony or regular scVI models, SysVI excels at aligning technical systems while preserving true biological variance, producing embedding spaces where analogous cell types across batches cluster coherently.</p><p>in the <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/scrna/sysVI.html" target="_blank" rel="noopener noreferrer">tutorial</a>, we show the power of sysVI with data integration between human and mouse immune cells.</p><img alt="sysvi" width="100%" src="/img/blog-post-scvi-tools-1p3/sysvi.png"><p>Figure 3: Example results of integration between human and mouse immune cells </p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="decipher"></a>Decipher<a class="hash-link" href="#decipher" title="Direct link to heading">#</a></h3><p>Decipher<sup id="fnref-6"><a href="#fn-6" class="footnote-ref">6</a></sup> Designed to dissect perturbation effects (e.g., disease versus control), Decipher disentangles shared and condition-specific variation within a VAE framework. <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/scrna/decipher_tutorial.html" target="_blank" rel="noopener noreferrer">Demonstrated</a> on AML (acute myeloid leukemia) data, it uncovers latent axes aligning with known disease signatures and identifies corresponding differentially expressed markers - bridging latent space analysis and functional biology.</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="methylvi"></a>MethylVI<a class="hash-link" href="#methylvi" title="Direct link to heading">#</a></h3><p>MethylVI<sup id="fnref-7"><a href="#fn-7" class="footnote-ref">7</a></sup> is a VAE tailored for single-cell bisulfite sequencing (scBS‑seq). By modeling methylation probabilities at genomic regions, it captures epigenetic heterogeneity and learns latent spaces that integrate multiple batches. <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/scbs/MethylVI_batch.html" target="_blank" rel="noopener noreferrer">Tutorial</a> show it outperforms linear methods like PCA in retaining biologically meaningful methylation structures.</p><img alt="methylvi" width="100%" src="/img/blog-post-scvi-tools-1p3/methylvi.png"><p>Figure 4: MethylVI integration of cell types from different single-cell bi-sulfite sequencing platforms</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="methylanvi"></a>MethylANVI<a class="hash-link" href="#methylanvi" title="Direct link to heading">#</a></h3><p>MethylANVI extends the MethylVI framework with annotation-aware modeling: it jointly integrates methylation profiles with metadata-driven cell-type labels. This supervised model supports both clustering and label transfer, all while capturing latent biological variation across methylome profiles, ideal for epigenetic atlas-building.</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="totalanvi"></a>totalANVI<a class="hash-link" href="#totalanvi" title="Direct link to heading">#</a></h3><p>totalANVI brings supervised annotation to CITE‑seq-style multi-omic integration. Leveraging a VAE for RNA + protein, it jointly learns latent embeddings and cell-type classifiers. The model simultaneously performs dimensionality reduction, denoising, differential expression, and accurate cell-type annotation in one cohesive model.</p><p>Tutorial in scvi-tools soon to be updated.</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="mrvi-in-pytorch"></a>mrVI in pyTorch<a class="hash-link" href="#mrvi-in-pytorch" title="Direct link to heading">#</a></h3><p>MrVI<sup id="fnref-8"><a href="#fn-8" class="footnote-ref">8</a></sup> (Multi-resolution Variational Inference) is a model written in <a href="https://docs.jax.dev/en/latest/notebooks/thinking_in_jax.html" target="_blank" rel="noopener noreferrer">Jax</a> for analyzing multi-sample, multi-batch single-cell RNA-seq data. MrVI is particularly suited for single-cell RNA sequencing datasets with comparable observations across many samples. It conducts both exploratory analyses (locally dividing samples into groups based on molecular properties) and comparative analyses (comparing pre-defined groups of samples in terms of differential expression and differential abundance) at single-cell resolution. </p><p>In recent scvi-tools releases we added a Pytorch implementation of mrVI, to be along with the Jax one, to support broader options on how to use this popular model. </p><p>Tutorial of mrVI in torch running on a subset of <a href="https://doi.org/10.1101/2025.02.20.639398" target="_blank" rel="noopener noreferrer">Tahoe100M</a> cells dataset can be found <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/scrna/Tahoe100_mrVI.html" target="_blank" rel="noopener noreferrer">here</a></p><img alt="lamin" width="100%" src="/img/blog-post-scvi-tools-1p3/mrvi.png"><p>Figure 5: mrVI Integration on Covid dataset</p><hr><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="2-🧩-custom-dataloaders"></a>2. 🧩 Custom Dataloaders<a class="hash-link" href="#2-🧩-custom-dataloaders" title="Direct link to heading">#</a></h2><p>scvi‑tools v1.4 introduces three scalable custom dataloaders: LaminDB, Census, and AnnCollection, enabling out-of-core and federated training without memory overload.
Custom Dataloaders are only supported in SCVI &amp; SCANVI models, but it should be easy to expand them to other models.
These backends support full compatibility with scvi‑tools data registration and training workflows, offering both scale and convenience to large projects.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="lamindb"></a>LaminDB<a class="hash-link" href="#lamindb" title="Direct link to heading">#</a></h3><p>Integrates with <a href="https://lamin.ai/" target="_blank" rel="noopener noreferrer">Lamindb</a>, enabling out-of-core training from disk-backed collections. Users can register collections and seamlessly train models like SCVI using lamin&#x27;s MappedCollection, benefiting from disk efficiency while maintaining full API compatibility with in-memory datasets. For more information see this <a href="https://docs.scvi-tools.org/en/stable/user_guide/use_case/custom_dataloaders.html" target="_blank" rel="noopener noreferrer">link</a>.</p><p>The next <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/custom_dl/lamin.html" target="_blank" rel="noopener noreferrer">tutorial</a> shows demonstration of a scalable approach to training an scVI model on PBMC data using Lamin dataloader.</p><img alt="lamin" width="100%" src="/img/blog-post-scvi-tools-1p3/lamin.png"><p>Figure 6: SCVI Integration achieved using LaminDB dataloader, on 2 distinct PBMC data.</p><p>This <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/custom_dl/Tahoe100_mrVI_lamin.html" target="_blank" rel="noopener noreferrer">tutorial</a> shows the analysis of mrVI in its PyTorch version together with Lamin Custom dataloader over a subset of Tahoe100M cells dataset.</p><img alt="lamin" width="100%" src="/img/blog-post-scvi-tools-1p3/lamin_tahoe100.png"><p>Figure 7: SCVI (bottom) &amp; MRVI (top) Integration achieved using LaminDB dataloader, on a subset of Tahoe100M cells data.</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="census"></a>Census<a class="hash-link" href="#census" title="Direct link to heading">#</a></h3><p>employs <a href="https://www.tiledb.com/" target="_blank" rel="noopener noreferrer">TileDB-SOMA</a> for atlas-scale tensor-backed data, offering similar streaming capabilities but enhanced support for multidimensional genomic inputs and federated study designs.
This custom dataloader directly read cellXgene dataset from S3 and train the SCVI model without the need to first download it, thus very suitable for few shots learning.</p><p>The next <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/custom_dl/tiledb.html" target="_blank" rel="noopener noreferrer">tutorial</a> shows demonstration of a scalable approach to training an scVI model on mus_musculus data using the Census dataloader</p><img alt="census" width="100%" src="/img/blog-post-scvi-tools-1p3/census.png"><p>Figure 8: SCVI Cell Integration achieved using Census dataloader, based on 4 type of batches: dataset_id, donor_id, assay and tissue_general</p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="anncollection"></a>AnnCollection<a class="hash-link" href="#anncollection" title="Direct link to heading">#</a></h3><p>This dataloader allows training on multiple AnnData objects simultaneously, without merging them into one dataset. AnnCollection handles disparities in features or layers internally and aligns them during training, empowering federated or multi-study analyses</p><p>The next <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/custom_dl/ann_collection.html" target="_blank" rel="noopener noreferrer">tutorial</a> shows how to apply the annCollection wrapper in scvi-tools to load and train SCANVI model on several adata&#x27;s that are stored on disk.
Another <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/hub/Tahoe100.html" target="_blank" rel="noopener noreferrer">link</a> shows how the <a href="https://doi.org/10.1101/2025.02.20.639398" target="_blank" rel="noopener noreferrer">Tahoe100M</a> cells dataset was trained in SCVI using the annCollection wrapper and its minified version was stored on scvi-hub for further analysis.</p><hr><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="3-️-core-enhancements"></a>3. ⚙️ Core Enhancements<a class="hash-link" href="#3-️-core-enhancements" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="multigpu-training"></a>Multi‑GPU Training<a class="hash-link" href="#multigpu-training" title="Direct link to heading">#</a></h3><p>Built on PyTorch Lightning, v1.4 empowers all major models to run across multiple GPUs with a single API flag. Training benchmarks times reduced by number of GPU exists, with full gradient synchronization and no code modifications needed.
See the following <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/use_cases/multiGPU.html" target="_blank" rel="noopener noreferrer">tutorial</a> and info <a href="https://docs.scvi-tools.org/en/stable/user_guide/use_case/multi_gpu_training.html" target="_blank" rel="noopener noreferrer">page</a></p><img alt="multi-gpu" width="100%" src="/img/blog-post-scvi-tools-1p3/multigpu.png"><p>Figure 9: comparison of SCVI training time between single and X2 multi-GPU machines as data increase. </p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="scibmetrics-optimization"></a>scIB‑Metrics Optimization<a class="hash-link" href="#scibmetrics-optimization" title="Direct link to heading">#</a></h3><p>With the integration of <code>ScibCallback</code> and the <code>AutotuneExperiment</code> class, users can now monitor <a href="https://github.com/YosefLab/scib-metrics" target="_blank" rel="noopener noreferrer">scIB metrics</a> on the validation set during training and automatically tune hyperparameters (model, training and architecture parameters) based on these metrics—directly optimizing for clustering and batch mixing performance, making the model training process more principled and outcome-driven.
See the following <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/use_cases/autotune_scvi.html#tuning-over-integration-metrics-with-scib-metrics" target="_blank" rel="noopener noreferrer">tutorial</a></p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="explainability--interpretability"></a>Explainability &amp; Interpretability<a class="hash-link" href="#explainability--interpretability" title="Direct link to heading">#</a></h3><p>v1.4 brings native support for <a href="https://captum.ai/api/integrated_gradients.html" target="_blank" rel="noopener noreferrer">Captum&#x27;s</a> Integrated Gradients (IG) across semi-supervised generative models like Scanvi and totalANVI. Users can compute marker-level attribution scores tied to latent dimensions or differential axes. This complements <code>get_normalized_expression</code>, delivering a full pipeline that links model representations back to biologically interpretable molecular mechanisms, offering transparency and interpretability in deep generative modeling</p><p>See the following <a href="https://docs.scvi-tools.org/en/stable/tutorials/notebooks/use_cases/interpretability.html" target="_blank" rel="noopener noreferrer">tutorial</a> as an example of scanvi model ran on a PBMC dataset from 10X.</p><img alt="ig" width="100%" src="/img/blog-post-scvi-tools-1p3/ig.png"><p>Figure 10: Integrated gradients total contribution per gene per cell type, over data of PBMC.</p><hr><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="summary"></a>Summary<a class="hash-link" href="#summary" title="Direct link to heading">#</a></h2><p>scvi‑tools v1.4 is a landmark release that advances the field across three foundational pillars:</p><ol><li><p><strong>Innovative modeling</strong> across ten tailored VAEs for spatial, protein, methylation, perturbation, and multi‑omic data.</p></li><li><p><strong>Scalable data processing</strong> with three new custom dataloaders in the backend, enabling efficient handling of federated, out-of-core, atlas-scale datasets, such as the Tahoe100M cells.</p></li><li><p><strong>Infrastructure and transparency</strong> with multi-GPU training, metric-aware tuning, and demonstrable model interpretability.</p></li></ol><p>Together, these developments empower researchers to build, train, and interpret probabilistic models at scale-in a reproducible, transparent, and biologically meaningful way.</p><hr><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="references"></a>References<a class="hash-link" href="#references" title="Direct link to heading">#</a></h2><div class="footnotes"><hr><ol><li id="fn-1">ResolVI: addressing noise and bias in spatial transcriptomics / Ergen et al.<a href="#fnref-1" class="footnote-backref">↩</a></li><li id="fn-2">scVIVA: a probabilistic framework for representation of cells and their environments in spatial transcriptomics / Levy et al.<a href="#fnref-2" class="footnote-backref">↩</a></li><li id="fn-3">CytoVI: Deep generative modeling of antibody-based single cell technologies / Ingelfinger et al.<a href="#fnref-3" class="footnote-backref">↩</a></li><li id="fn-4">VI-VS: calibrated identification of feature dependencies in single-cell multiomics / Boyeau et al.<a href="#fnref-4" class="footnote-backref">↩</a></li><li id="fn-5">sysVI: Integrating single-cell RNA-seq datasets with substantial batch effects / Hrovatin et al.<a href="#fnref-5" class="footnote-backref">↩</a></li><li id="fn-6">Decipher: Joint representation and visualization of derailed cell states with Decipher / Nazaret et al.<a href="#fnref-6" class="footnote-backref">↩</a></li><li id="fn-7">MethylVI: A deep generative model of single-cell methylomic data / Weinberger et al.<a href="#fnref-7" class="footnote-backref">↩</a></li><li id="fn-8">MrVI: Deep generative modeling of sample-level heterogeneity in single-cell genomics / Boyeau et al.<a href="#fnref-8" class="footnote-backref">↩</a></li></ol></div></div><footer class="row docusaurus-mt-lg"><div class="col col--9"><b>Tags:</b><ul class="tags_2ga9 padding--none margin-left--sm"><li class="tag_11ep"><a class="tag_1Okp tagRegular_3MiF" href="/blog/tags/scvi-tools">scvi-tools</a></li><li class="tag_11ep"><a class="tag_1Okp tagRegular_3MiF" href="/blog/tags/release">release</a></li></ul></div><div class="col col--3 text--right"><a aria-label="Read more about scvi-tools 1.4 release" href="/blog/v13"><b>Read More</b></a></div></footer></article><article class="margin-bottom--xl" itemprop="blogPost" itemscope="" itemtype="http://schema.org/BlogPosting"><header><h2 class="blogPostTitle_GeHD" itemprop="headline"><a itemprop="url" href="/blog/v090">scvi-tools 0.9.0 release</a></h2><div class="blogPostData_291c margin-vert--md"><time datetime="2021-03-03T00:00:00.000Z" itemprop="datePublished">March 3, 2021</time> · 7 min read</div><div class="row margin-top--md margin-bottom--sm"><div class="col col--6 authorCol_1R69"><div class="avatar margin-bottom--sm"><div class="avatar__intro" itemprop="author" itemscope="" itemtype="https://schema.org/Person"><div class="avatar__name"><a itemprop="url"><span itemprop="name">Adam Gayoso, Romain Lopez, Galen Xing, Nir Yosef</span></a></div></div></div></div></div></header><div class="markdown" itemprop="articleBody"><p>Today we officially released <code>scvi-tools</code> version 0.9.0 (<a href="https://docs.scvi-tools.org/en/stable/release_notes/index.html" target="_blank" rel="noopener noreferrer">changelog</a>). This release marks the culmination of five months of work on the backend of the codebase, which came after three months of work on the frontend.
In this short note, we officially introduce <code>scvi-tools</code> as a readily usable codebase that contains many implementations of probabilistic single-cell omics methods, and also features a high-level interface to accelerate the model development process. We start with some historical notes about our previous codebase, which was mostly used for internal developments in the last three years. We then describe the obstacles we found to its external adoption, and the foundational idea behind the new <code>scvi-tools</code> work: a high-level deep probabilistic programming library specialized for single-cell omics data.</p></div><footer class="row docusaurus-mt-lg"><div class="col col--9"><b>Tags:</b><ul class="tags_2ga9 padding--none margin-left--sm"><li class="tag_11ep"><a class="tag_1Okp tagRegular_3MiF" href="/blog/tags/scvi-tools">scvi-tools</a></li><li class="tag_11ep"><a class="tag_1Okp tagRegular_3MiF" href="/blog/tags/release">release</a></li></ul></div><div class="col col--3 text--right"><a aria-label="Read more about scvi-tools 0.9.0 release" href="/blog/v090"><b>Read More</b></a></div></footer></article></main></div></div></div><footer class="footer footer--dark"><div class="container"><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2025, Yosef Lab, Weizmann Institute of Science. Built with Docusaurus.</div></div></div></footer></div>
<script src="/assets/js/runtime~main.25fa66c4.js"></script>
<script src="/assets/js/main.5868d4e3.js"></script>
</body>
</html>